### PR TITLE
fix(files_versions): Migrate version name dialog from NcModal to NcDialog

### PR DESCRIPTION
**Pull request by susnux**
* Resolves https://github.com/nextcloud/viewer/issues/2390

## Summary
Make the version name dialog a real dialog instead of a modal.

### Screen recording (with this PR)
[Bildschirmaufnahme_20240822_013252.webm](https://github.com/user-attachments/assets/51f3d27b-acae-4521-95ab-22a2026d6e90)

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
From a UX perspective, if the version isn't a named version yet - than the re-destructive action button of removing the version name seem misleading.

**Review by artonge — APPROVED** [review]

File: apps/files_versions/src/components/VersionLabelDialog.vue [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	},
+	computed: {
+		dialogButtons() {
+			return [
+				{
+					label: t('files_versions', 'Remove version name'),
+					type: 'error',
+					nativeType: 'reset',
+					callback: () => { this.setVersionLabel('') },
+				},
```
Comment (artonge) [inline apps/files_versions/src/components/VersionLabelDialog.vue:65]: Previously there was a condition on that button:

```vue
	<NcButton :disabled="innerVersionLabel.trim().length === 0" @click="setVersionLabel('')">
		{{ t('files_versions', 'Remove version name') }}
	</NcButton>
```

Comment (susnux) [reply apps/files_versions/src/components/VersionLabelDialog.vue:65]: :see_no_evil: 

Fixed

**Comment by susnux** [comment]
> From a UX perspective, if the version isn't a named version yet - than the re-destructive action button of removing the version name seem misleading.

Makes sense, adjusted:
[Bildschirmaufnahme_20240822_135654.webm](https://github.com/user-attachments/assets/75180771-7adc-46c5-95ad-ca9e460591f4)

**Review by AndyScherzinger — APPROVED** [review]

**Pull request merged**